### PR TITLE
[iOS] Fix Linker Errors On Build

### PR DESCRIPTION
**Pull request by erandagan**
Resolves https://github.com/callstack/react-native-fbads/issues/144

This PR resolves the linker errors in the current release (5x). The above issue demonstrates the problem, which I experienced myself in my project.

The error originates from changes to the xcode configuration with some incorrect values, including header & framework paths, code signing (which was linked to @Suraj-Tiwari own profile I believe) as well as the `MACH_O_TYPE` value.

I verified the problem has been resolved - projects now compile correctly against this library.

### CODE REVIEW

File: ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -240,8 +240,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 11.4;
-				MACH_O_TYPE = staticlib;
+				IPHONEOS_DEPLOYMENT_TARGET = 7.0;
```
Comment (Suraj-Tiwari) [inline ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj]: Please set 8.0 as deployment target.

File: ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -293,8 +292,7 @@
...
```
Comment (Suraj-Tiwari) [inline ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj]: Deployment target 8.0

File: ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -304,26 +302,38 @@
 		D2AFD33D211B138E00580DF0 /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				CODE_SIGN_STYLE = Automatic;
-				DEVELOPMENT_TEAM = W3VSH3LGL7;
-				FRAMEWORK_SEARCH_PATHS = "$(PROJECT_DIR)/../../../ios/Pods/FBAudienceNetwork/";
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)",
+					"$(PROJECT_DIR)/**",
```
Comment (Suraj-Tiwari) [inline ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj]: Not good.
remove 
$(PROJECT_DIR)/** (there is no framework included) 
set `$(PROJECT_DIR)/../../../ios/Pods/FBAudienceNetwork/`

File: ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -304,26 +302,38 @@
...
+					"$(SRCROOT)/../../../ios/Pods/FBAudienceNetwork",
+				);
+				HEADER_SEARCH_PATHS = (
```
Comment (Suraj-Tiwari) [inline ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj]: "HEADER_SEARCH_PATHS" required?

File: ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj (hunk 3 above)
Comment (erandagan) [reply ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj]: This is for the case of compiling the library on its own while testing.

File: ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj (hunk 4 above)
Comment (erandagan) [reply ios/ReactNativeAdsFacebook.xcodeproj/project.pbxproj]: Yes

**Comment by erandagan** [comment]
@Suraj-Tiwari have a look now

**Pull request merged**